src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ResourceDataNode.cs

```diff
@@ -0,0 +1,182 @@
+        /// </summary>
+        public int EcmaOffset { get; }
+
+        /// <summary>
+        /// Module the resource is defined in
+        /// </summary>
+        public EcmaModule EcmaModule { get; }
+
+        /// <summary>
+        /// Length of the resource
+        /// </summary>
+        public int Length { get; }
+    }
+
+    /// <summary>
+    /// Blob of data containing resources for all assemblies generated into the image.
+    /// Resources are simply copied from the inputs and concatenated into this blob.
+    /// All format information is provided by <see cref="ResourceIndexNode"/>
+    /// </summary>
+    internal class ResourceDataNode : ObjectNode, ISymbolNode
+    {
+        private HashSet<ModuleDesc> _modulesSeen;
+
+        /// <summary>
+        /// Resource index information generated while extracting resources into the data blob
+        /// </summary>
+        public List<ResourceIndexData> IndexData { get; } = new List<ResourceIndexData>();
+
+        public ResourceDataNode(HashSet<ModuleDesc> modulesSeen)
+        {
+            _modulesSeen = modulesSeen;
+            _endSymbol = new ObjectAndOffsetSymbolNode(this, 0, "__embedded_resourcedata_End", true);
+        }
+
+        private ObjectAndOffsetSymbolNode _endSymbol;
+        public ISymbolNode EndSymbol => _endSymbol;
+
+        public override bool IsShareable => false;
+
+        public override ObjectNodeSection Section => ObjectNodeSection.ReadOnlyDataSection;
+
+        public override bool StaticDependenciesAreComputed => true;
+
+        public int Offset => 0;
+
+        public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
+        {
+            sb.Append(nameMangler.CompilationUnitPrefix).Append("__embedded_resourcedata");
+        }
+
+        protected override string GetName() => this.GetMangledName();
+
+        public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
+        {
+            // This node has no relocations.
+            if (relocsOnly)
+                return new ObjectData(Array.Empty<byte>(), Array.Empty<Relocation>(), 1, new ISymbolNode[] { this });
+            
+            byte[] blob = GenerateResourceBlob(factory.CompilationModuleGroup, factory.TypeSystemContext);
+            return new ObjectData(
+                blob,
+                Array.Empty<Relocation>(),
+                1,
+                new ISymbolNode[]
+                {
+                                this,
+                                EndSymbol
+                });
+        }
+
+        /// <summary>
+        /// Extracts resources from all modules being compiled into a single blob and saves
+        /// the information needed to create an index into that blob.
+        /// </summary>
+        private byte[] GenerateResourceBlob(CompilationModuleGroup compilationModuleGroup, CompilerTypeSystemContext typeSystemContext)
```

Comment:
Removed (I suspect they might come back when we refine deciding which assemblies get resources).